test_vector_search_in_cloud.py

```diff
@@ -0,0 +1,274 @@
+import ast
+import random
+import time
+from functools import cached_property
+from threading import Thread
+from typing import Optional
+
+import cassandra
+
+from sdcm.mgmt.helpers import get_schema_create_statements_from_file, substitute_dc_name_in_ks_statements_if_different
+from sdcm.tester import ClusterTester
+from sdcm.utils.common import S3Storage
+from sdcm.utils.file import File
+
+# Vector Search test data files
+BUCKET_NAME = "vector-store-in-cloud"
+
+TEST_DATA_FILENAME = "test_data.txt"
+GROUND_TRUTH_FILENAME = "ground_truth.txt"
+DATASET_FILENAME = "dataset.txt"
+SCHEMA_FILENAME = "schema.json"
+
+DATA_DIR_PATH = "data_dir/vector_search/"
+TEST_DATA_FILE_PATH = DATA_DIR_PATH + TEST_DATA_FILENAME
+GROUND_TRUTH_FILE_PATH = DATA_DIR_PATH + GROUND_TRUTH_FILENAME
+DATASET_FILE_PATH = DATA_DIR_PATH + DATASET_FILENAME
+SCHEMA_FILE_PATH = DATA_DIR_PATH + SCHEMA_FILENAME
+
+ANN_SEARCH_CMD_TEMPLATE = "SELECT id FROM vdb_bench.vdb_bench_collection ORDER BY vector ANN OF {vector} LIMIT 10;"
+
+VECTOR_SEARCH_INSTANCE_TYPES = {
+    "aws": {"t4g.small": 175, "t4g.medium": 176, "r7g.medium": 177},
+    "gce": {"e2-small": 178, "e2-medium": 179, "n4-highmem-2": 180}
+}
+
+
+class VectorSearchInCloudBase(ClusterTester):
+    average_recall: float = 0.0
+
+    def get_email_data(self):
+        self.log.info("Prepare data for email")
+
+        email_data = self._get_common_email_data()
+        email_data.update({"average_recall": self.average_recall})
+
+        return email_data
+
+    @staticmethod
+    def download_vector_search_test_data_from_s3():
```

Comment:
I would recommend putting this into the same place we have other sstable, we already have the code for downloading the files, and putting loading them into scylla, is should be duplicated

Reply:
https://github.com/scylladb/scylla-cluster-tests/blob/c8c8487db5ec5adc77e2bc5b2859566d5a7d7ede/sdcm/utils/sstable/load_utils.py#L19